rinex/tests/parsing.rs

```diff
@@ -42,6 +43,89 @@ mod test {
+                                                ion_msg.as_bdgim().is_some() == true,
+                                                "BeiDou (CNVX) should be interpreted as Bd model"
+                                            );
+                                        },
+                                        _ => {
+                                            panic!(
+                                                "invalid message type \"{}\" for BeiDou ION frame",
+                                                msg
+                                            );
+                                        },
+                                    }
+                                } else if sv.constellation == Constellation::IRNSS {
+                                    assert!(
+                                        ion_msg.as_klobuchar().is_some(),
+                                        "only Kb models provided by NavIC/IRNSS vehicles"
+                                    );
+                                } else if sv.constellation == Constellation::Galileo {
+                                    assert!(
+                                        ion_msg.as_nequick_g().is_some(),
+                                        "only Ng models provided by GAL vehicles"
+                                    );
+                                } else {
+                                    panic!(
+                                        "incorrect constellation provider of an ION model: {}",
+                                        sv.constellation
+                                    );
+                                }
+                            }
+                            /*
+                             * Verify EOP logical correctness
+                             */
+                            for (_, (msg, sv, _)) in rinex.earth_orientation() {
+                                assert!(
+                                    (sv.constellation == Constellation::GPS)
+                                    || (sv.constellation == Constellation::QZSS)
+                                    || (sv.constellation == Constellation::IRNSS)
+                                    || (sv.constellation == Constellation::BeiDou), "constellation \"{}\" not declared as EOP frame provider, according to V4 specs", sv.constellation);
+                                assert!(
+                                    (msg == NavMsgType::CNVX) || (msg == NavMsgType::LNAV),
+                                    "bad msg identified for GPS vehicle: {}",
+                                    msg
+                                );
+                            }
+                            /*
+                             * Verify STO logical correctness
+                             */
+                            for (_, (msg, sv, _)) in rinex.system_time_offset() {
+                                assert!(
+                                    (msg == NavMsgType::LNAV)
```

Comment:
`matches!` works here, too.

Reply:
applied, but in the following form,  
because I like having meaningful panic messages that help debug what went wrong.
`matches!` does not allow for a panic message

```rust
match var {
    cond1 | cond2 | cond3 => {}, 
    _ => panic!("oops"),
}
```

Reply:
You can use a `match` like in your snippet, or keep doing `assert!(matches!(..., ...), ...)`.

Reply:
Oh I see, assert! on matches! which is then a boolean condition. Will try next time